packages/core/src/components/templates/ProductListingPage/ProductListingPage.tsx

```diff
@@ -70,69 +81,95 @@ const useSearchParams = ({
   return useMemo(() => parseSearchState(new URL(hrefState)), [hrefState])
 }
 
+const overwriteMerge = (_, sourceArray, __) => sourceArray
+
 export default function ProductListingPage({
   page: { sections, settings },
-  ...otherProps
+  data: server,
 }: ProductListingPageProps) {
-  const { collection } = otherProps
+  const collection = server.collection
   const router = useRouter()
   const applySearchState = useApplySearchState()
   const searchParams = useSearchParams({
-    ...otherProps,
+    collection,
     sort: settings?.productGallery?.sortBySelection as SearchState['sort'],
   })
 
-  const { page, sort } = searchParams
+  const { page, sort, term, selectedFacets } = searchParams
   const title = collection?.seo.title ?? storeConfig.seo.title
   const description = collection?.seo.description ?? storeConfig.seo.title
   const pageQuery = page !== 0 ? `?page=${page}` : ''
   const separator = pageQuery !== '' ? '&' : '?'
   const sortQuery = !!sort ? `${separator}sort=${sort}` : ''
   const [pathname] = router.asPath.split('?')
   const canonical = `${storeConfig.storeUrl}${pathname}${pageQuery}${sortQuery}`
+  const itemsPerPage = settings?.productGallery?.itemsPerPage ?? ITEMS_PER_PAGE
+  const initialPage = useRef(page)
+
+  const { data: pageProductGalleryData } = useProductGalleryQuery({
+    term,
+    sort,
+    selectedFacets,
+    itemsPerPage,
+  })
+
+  const { productsPerPage } = usePageProductsQuery({
```

Comment:
Does this and line 109 mean that we are doing two requests to the server?

Reply:
2 requests using the `search` query and bringing different data. Are you thinking about grouping this into 1 single request? I'm also thinking about that. Let's do some experiments. I'm not 100% convinced because we reuse the `usePageProductsQuery` logic in the `ProductGalleryPage`.

Reply:
I think it's worth trying as well. We can have a different logic at the page level, I don't think that would be an issue.